src/compiler/binder.ts

```diff
@@ -1297,6 +1297,9 @@ namespace ts {
                     // as other properties in the object literal.  So we use SymbolFlags.PropertyExcludes
                     // so that it will conflict with any other object literal members with the same
                     // name.
+                    if (options.strictThis) {
+                        seenThisKeyword = true;
+                    }

```

Comment:
I ran some numbers this afternoon and didn't see a big change. Could be misreading the results though.